### PR TITLE
Upgrade Ruby to 3.2

### PR DESCRIPTION
**Pull request by gclssvglx**
Ruby v2.7.6 is EOL on 31 March 2023, so we need to update to a later version. 

This change:

- Updates the version of Ruby to v3.2(.0)
- Fixes linter issues generated by the update above
- Fixes an issue when updating Psych to >= v3.1.0
- Fixes an issue when updating RSpec-Mocks to >= v3.10.3

[Trello](https://trello.com/c/Ih8jw0Hw/365-update-to-ruby-32)

### CODE REVIEW

File: lib/content_item_publisher/content_item_presenter.rb [hunk 1]
```diff
@@ -27,7 +27,7 @@ def present
     end
 
     def present_links
-      { content_id: content_id, links: {} }
+      { content_id:, links: {} }
```
Comment (Rosa-Fox) [inline lib/content_item_publisher/content_item_presenter.rb:30]: I'm confused about how this is working... does it just automatically use a value with the same name as the key if its not specified?

**Review by Rosa-Fox — APPROVED** [review]

File: lib/content_item_publisher/content_item_presenter.rb (hunk 1 above)
Comment (gclssvglx) [reply lib/content_item_publisher/content_item_presenter.rb:30]: Yeah, exactly right! I find it kinda confusing to be honest. But, it does clear up the params nicely though.

**Comment by gclssvglx** [comment]
Thanks Rosa

File: lib/content_item_publisher/content_item_presenter.rb (hunk 1 above)
Comment (csutter) [reply lib/content_item_publisher/content_item_presenter.rb:30]: Whoever thought we'd be borrowing syntactic sugar from Javascript!

Also thanks for sorting this out - I dreaded having to do it myself as part of the search improvement work ❤️

**Pull request merged**